good_smell/smells/yield_from.py

```diff
@@ -0,0 +1,40 @@
+from good_smell import AstSmell, LoggingTransformer
+import ast
+
+
+class YieldFrom(AstSmell):
+    """Checks for yields inside for loops"""
+
+    @property
+    def transformer_class(self):
+        return YieldFromTransformer
+
+    @property
+    def warning_message(self):
+        return "Consider using yield from instead of yield inside of a for loop"
+
+    @property
+    def symbol(self):
+        return "yield-from"
+
+
+class YieldFromTransformer(LoggingTransformer):
+    """NodeTransformer that goes visits all the nested `for`s and replaces them
+    with itertools.product"""
+
+    def visit_For(self, node: ast.For):
+        yield_from = ast.Expr(
+            value=ast.YieldFrom(node.iter)
+        )
+        return ast.fix_missing_locations(yield_from)
+
+    @staticmethod
+    def is_smelly(node: ast.AST):
+        """Check if the node is only a nested for"""
+        # TODO check python version, check if inside a function
```

Comment:
Yielding outside a function results in a SyntaxError, fixing that is outside this projects' scope. So I think it's valid to assume that if you see a yield, it's inside a function

Reply:
Please check python version by adding a function to LintSmell which checks if it should run at all - Adding a check in `is_smelly` will run the check for _every node_ which is redundant